datafusion/proto/src/logical_plan/to_proto.rs

```diff
@@ -352,6 +352,11 @@ impl From<&StringifiedPlan> for protobuf::StringifiedPlan {
                 PlanType::FinalPhysicalPlan => Some(protobuf::PlanType {
                     plan_type_enum: Some(FinalPhysicalPlan(EmptyMessage {})),
                 }),
+                // Make it `todo` to avoid breaking clippy
+                // We do not want to add proto plan type for these two becasue they are just the same as
+                // InitialPhysicalPlan and FinalPhysicalPlan
+                datafusion_expr::PlanType::InitialPhysicalPlanWithStats
+                | datafusion_expr::PlanType::FinalPhysicalPlanWithStats => todo!(),
```

Comment:
@alamb Cargo suggested me to do this. Do you want to go with this or create 2 new types in photo enum?

Reply:
I think adding two new types in the proto would be the most consistent with the existing code

However, as long as this doesn't panic I think it would be ok to merge -- for example, if it returned an internal error. 

I don't really understand why there is special serialization for this to be honest 🤔  maybe we could simplify that too as a follow on PR

Reply:
@alamb  The `todo!`  returns a panic of `not yet implemented`.  Let me try to add new types to proto

Reply:
I have add 2 new types to proto instead